common/school/externalSearch.ts

```diff
@@ -0,0 +1,61 @@
+import { school_state_enum as State, pupil_schooltype_enum as SchoolType } from '@prisma/client';
+import { getLogger } from '../logger/logger';
+import { SchoolTypeMap } from './schoolType';
+
+export interface ExternalSchool {
+    id: string;
+    name: string;
+    city?: string;
+    school_type?: string;
+    zip?: string;
+    email?: string;
+}
+
+export interface SchoolResult extends Omit<ExternalSchool, 'school_type'> {
+    schooltype?: SchoolType;
+    state: State;
+}
+
+const logger = getLogger('Jedeschule');
+
+const getStateFromExternalSchool = (school?: ExternalSchool) => {
+    const state = school?.id.substring(0, 2).toLowerCase();
+    const isValid = Object.values(State).some((e) => e === state);
+    if (!isValid) {
+        /**
+         * The state is always the first part of the school id, if for some reason the state doesn't match
+         * we need the heads up about that.
+         */
+        const msg = 'Could not get state from school';
+        logger.error(msg, new Error(msg), school);
```

Comment:
(not so much a fan of non-error handling ... why not just throw an error here?)

Reply:
I'm handling the potential "error" with the validation. I don't want the whole endpoint to fail due to a single corrupted result. The Frontend can still work with the other results (and even with this one by not skipping the state step).